docs/DeveloperGuide.md

```diff
@@ -1,29 +1,134 @@
 # Developer Guide
 
+## Welcome!
+Welcome and thanks for joining us in building our program created to handle all of your currencies and money problems. With us you can use such functions as transfer, withdraw, add with different currencies to handle all your daily money problems. You also have the opportunity to add your own currencies, such as cryptocurrencies and etc. 
+
+Within this documentation you can find all the information needed to keep building it.
+
+## Setting up
+
+This section describes the development tools used in the creation of Currency Manager.
+
+### 2.1 Prerequisites
+  1. JDK 11
+  2. Intellij IDEA
+  3. Gradle 6.2
+
+### 2.2 Setting up the project
+  1. Fork this repo and clone the fork onto your computer
+  2. Open Intellij(if you are not in the welcome screen, click `File` > `Close Project` to close the existing project dialog first).
+  3. Set up the correct JDK version for Gradle
+      a) That can be done under `Configure` > `Project Defaults` > `Project Structure`
+  4. Click `Import Project`
+  5. Find the `build.gradle` file and select it. Click `OK`.
+  6. Click `Open as Project`
+  7. Click `OK` and accept the default settings.
+
 ## Acknowledgements
 
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
 ## Design & implementation
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### Save / Withdraw feature
+
+The save and withdraw feature is implemented based on the Wallet class data structure. 
+The Wallet class has several components: the username, the password, the default currency, the total balance, 
+and the deposit list containing different types of currency and their balances.
+
+Given below is the UML diagram of the Wallet class.
+
+![avatar](/property/WalletClass.jpg)
+
+Here is an example usage scenario and how the save/withdraw feature behaves at each step.
+
+Step 1: The user register a new account and a new `Wallet` object is initialized.
+For now, the deposit list is empty.
+
+![avatar](/property/ObjectStep1.jpg)
+
+Step 2: The user login and executes `save sgd 100` to save 100 singapore dollars in the wallet. 
+The `save` command first parses the command and verify its validity. 
+After that, it calls the `saveMoney` method of the `Wallet` class, 
+and the `saveMoney` method traverses the deposit list and finds that there's no existed SGD deposit. 
+Therefore, it creates a new `Deposit` object with the currency of SGD and the balance of 100, 
+and then adds it to the deposit list.
+Also, the total balance is updated.
+
+![avatar](/property/ObjectStep2.jpg)
```

Comment:
Maybe this can be the overall class diagram, with the content after the ":" being the data type rather than the example values. 
<img width="718" alt="image" src="https://user-images.githubusercontent.com/88488001/198248113-1d79f3d6-aa3f-480d-aba7-2896154d86e9.png">